api/cloudcontroller/ccv3/space.go

```diff
@@ -40,3 +45,34 @@ func (client *Client) GetSpaces(query ...Query) ([]Space, Warnings, error) {
 
 	return fullSpacesList, warnings, err
 }
+
+func (client *Client) UpdateSpace(space Space) (Space, Warnings, error) {
+	spaceGUID := space.GUID
+	space.GUID = ""
+	space.Relationships = nil
```

Comment:
Having to unset these feels weird to us and we saw this also came up in https://github.com/cloudfoundry/cli/commit/781f526de8c1f8b6a065ec18fd20fe0965125790:

> Some weirdness because we need to retrieve the guid field from v3 on the
organization entity, but we are not permitted to send it in a patch
request.

I wonder since these are patch requests if we could actually get away with just having a `MetadataPatchRequest` (or something) struct that all the endpoints could use instead of us having to delete things from each resource piecemeal. Wanted to get some of your thoughts on that before attempting it, though.

Reply:
Another thing I was thinking was what if we had a generic `ccv3/metadata.go` client. The only thing that's really going to be different between any of these resources is the endpoint that's called and we could probably just switch on `resourceName`.

They should all behave nearly the same otherwise -- especially if we limit the request struct to being purely scoped to metadata like I mentioned above. Thoughts?

Reply:
I like that idea. It would clean up a lot of this duplication.

Reply:
Logistically what do you think the best way to tackle that refactor would be? Would you be ok with me capturing this in a chore story that CAPI could prioritize working on next and PR separately or is this something y'all would like to cross-team pair on?

I feel like the refactor would make the most sense with these three `set-label` resources merged and in place as a foundation for it .

Reply:
Follow-up PR: https://github.com/cloudfoundry/cli/pull/1652